libmisc/salt.c

```diff
@@ -34,7 +34,7 @@ static /*@observer@*/const char *BCRYPT_salt_rounds (/*@null@*/int *prefered_rou
 #endif /* USE_BCRYPT */
 #ifdef USE_YESCRYPT
 static /*@observer@*/const char *gensalt_yescrypt (void);
-static /*@observer@*/const char *YESCRYPT_salt_cost (/*@null@*/long *prefered_rounds);
+static /*@observer@*/const char *YESCRYPT_salt_cost (/*@null@*/int *prefered_cost);
```

Comment:
I guess it's lucky this didn't appear in a release, since it would have been broken on big endian platforms... Shouldn't it be possible to fix the APIs so casting to and from `void *` to different width integer pointers isn't necessary?

(this isn't me raising an issue with this PR, just a comment)

Reply:
Could you please elaborate a bit more, why this would be broken on big endian platforms?

Reply:
I mean, *all* the other methods use a cast from `void *` to `int *` for the parameter…  Are they broken too in case of a big endian platform?

Reply:
I meant it was lucky the previous version didn't appear in a release. `long *` and `int *` are different widths on 64 bit platforms, and if the API users expected a pointer to `int`, interpreting as a pointer to `long *` would have read a completely different piece of memory. The big endian case would have mattered if API users passed `long *` and the function used `int *`, which is the opposite of the issue here; the previous version before your fix was always accessing memory it shouldn't.

> I mean, all the other methods use a cast from void * to int * for the parameter… Are they broken too in case of a big endian platform?

They aren't broken, but doing `void *` to `int *` instead of defining APIs to be `int *` makes it easy to let such errors pass by, since they can't be type checked.

Reply:
Now I get you.  =)